scipy/io/matlab/_mio4.py

```diff
@@ -171,18 +175,21 @@ def read_sub_array(self, hdr, copy=True):
             of dtype given by `hdr` ``dtype`` and shape given by `hdr` ``dims``
         '''
         dt = hdr.dtype
-        dims = hdr.dims
-        num_bytes = dt.itemsize
-        for d in dims:
-            num_bytes *= d
-        buffer = self.mat_stream.read(int(num_bytes))
+        # Fast product for large (>2GB) arrays.
+        num_bytes = reduce(mul, hdr.dims, np.int64(dt.itemsize))
+        if num_bytes > _MAX_INTP:
+            raise ValueError("Variable '" + hdr.name.decode('latin1') +
+                             "' has byte length longer than largest possible"
+                             " Numpy array on this platform")
```

Comment:
The `f`'s on the second and third lines are optional based on personal style preference.

```suggestion
            raise ValueError(f"Variable '{hdr.name.decode('latin1')}'"
                             f" has byte length longer than the largest possible"
                             f" NumPy array on this platform.")
```